integration/palette/src/main/java/com/bumptech/glide/integration/palette/PaletteBitmapViewTarget.java

```diff
@@ -17,4 +31,232 @@ public PaletteBitmapViewTarget(ImageView view) {
+         * such as background, text color, usually something visual and colorful. Must handle lack of a swatch.
+         */
+        public interface SwatchTarget {
+            void set(Palette.Swatch swatch);
+        }
+
+        public static final SwatchSelector VIBRANT = new SwatchSelector() {
+            public Palette.Swatch select(Palette palette) {
+                return palette.getVibrantSwatch();
+            }
+        };
+        public static final SwatchSelector VIBRANT_LIGHT = new SwatchSelector() {
+            public Palette.Swatch select(Palette palette) {
+                return palette.getLightVibrantSwatch();
+            }
+        };
+        public static final SwatchSelector VIBRANT_DARK = new SwatchSelector() {
+            public Palette.Swatch select(Palette palette) {
+                return palette.getDarkVibrantSwatch();
+            }
+        };
+
+        public static final SwatchSelector MUTED = new SwatchSelector() {
+            public Palette.Swatch select(Palette palette) {
+                return palette.getMutedSwatch();
+            }
+        };
+        public static final SwatchSelector MUTED_LIGHT = new SwatchSelector() {
+            public Palette.Swatch select(Palette palette) {
+                return palette.getLightMutedSwatch();
+            }
+        };
+        public static final SwatchSelector MUTED_DARK = new SwatchSelector() {
+            public Palette.Swatch select(Palette palette) {
+                return palette.getDarkMutedSwatch();
+            }
+        };
+
+        private final ImageView image;
+        private int defaultFallbackBackground;
+        private int defaultFallbackTextColor;
+
+        private final List<ActionGroup> actions = new LinkedList<ActionGroup>();
+        private ActionGroup current;
+
+        public Builder(ImageView image) {
+            this.image = image;
+            this.defaultFallbackBackground = Color.TRANSPARENT;
+            this.defaultFallbackTextColor = getDefaultTextColor(image.getContext());
+        }
+
+
+        public Builder defaultFallback(int fallbackBackground, int fallbackTextColor) {
+            this.defaultFallbackBackground = fallbackBackground;
+            this.defaultFallbackTextColor = fallbackTextColor;
+            return this;
+        }
+
+        public int getDefaultFallbackBackground() {
+            return defaultFallbackBackground;
+        }
+
+        public int getDefaultFallbackTextColor() {
+            return defaultFallbackTextColor;
+        }
+
+        public ImageView getImage() {
+            return image;
+        }
+
+        private void checkCurrent() {
+            if (current == null) {
+                throw new IllegalStateException("You must call .swatch() first.");
+            }
+        }
+
+        public Builder swatch(SwatchSelector selector, int fallbackBackground, int fallbackTextColor) {
```

Comment:
There are some new commits in https://github.com/TWiStErRob/glide/tree/palette-lib please check them out. I didn't want to open a 3rd PR for this until it's ready (can't change PRs branches and this new one is for 4.0).